### PR TITLE
Chat Sticker parsing and RequestItemUseHandler rework

### PR DESCRIPTION
**Pull request by Zintixx**
- Parsing Chat Stickers
- Added Verifying if user has a chat sticker group set
- Adding functionality for favoriting and unfavoriting chat stickers
- Parsing all function names from all items
- Reworked RequestItemUseHandler to support more than just boxes.
- Minor group cleanups to remove unused namespaces

### CODE REVIEW

File: GameDataParser/Parsers/ItemParser.cs [hunk 1]
```diff
@@ -206,6 +208,12 @@ protected override List<ItemMetadata> Parse()
                         }
                     }
                 }
+                if (contentType == "ChatEmoticonAdd")
+                {
+                    string rawParameter = function.Attributes["parameter"].Value;
+                    int parameter = int.Parse((rawParameter.Where(x => char.IsDigit(x)).ToArray())); // grab the proper value from parameter attribute
```
Comment (angelotadeucci) [inline GameDataParser/Parsers/ItemParser.cs:214]: What `.ToArray()` does here?

File: MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs [hunk 2]
```diff
@@ -14,64 +15,93 @@ public class RequestItemUseHandler : GamePacketHandler
...
+
+            if (item.Amount <= 1)
             {
-                return;
+                InventoryController.Remove(session, item.Uid, out Item removed);
+            }
+            else
+            {
+                item.Amount -= 1;
+                InventoryController.Update(session, item.Uid, item.Amount);
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs:58]: Use `InventoryController.Consume()` instead

File: MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs [hunk 3]
```diff
@@ -14,64 +15,93 @@ public class RequestItemUseHandler : GamePacketHandler
...
+                InventoryController.Remove(session, item.Uid, out Item removed);
             }
             // Decrement box amount to otherwise
             else
             {
-                box.Amount -= 1;
-                InventoryController.Update(session, boxUid, box.Amount);
+                item.Amount -= 1;
+                InventoryController.Update(session, item.Uid, item.Amount);
             }
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs:82]: Can you change to `Consume()` here too?

File: MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs [hunk 4]
```diff
@@ -14,64 +15,93 @@ public class RequestItemUseHandler : GamePacketHandler
...
+
+            if (item.Amount <= 1)
+            {
+                InventoryController.Remove(session, item.Uid, out Item removed);
+            }
+            else
+            {
+                item.Amount -= 1;
+                InventoryController.Update(session, item.Uid, item.Amount);
             }
```
Comment (angelotadeucci) [inline MapleServer2/PacketHandlers/Game/RequestItemUseHandler.cs:102]: And `Consume()` here too

**Review by angelotadeucci — CHANGES_REQUESTED** [review]

File: GameDataParser/Parsers/ItemParser.cs (hunk 1 above)
Comment (Zintixx) [reply GameDataParser/Parsers/ItemParser.cs:214]: Not really sure how else I would grab the value.

File: GameDataParser/Program.cs [hunk 5]
```diff
@@ -44,7 +44,7 @@ static async Task Main()
                 new AchieveParser(resources),
                 new RecipeParser(resources),
                 new MasteryParser(resources),
-                new NpcParser(resources)
```
Comment (AlanMorel) [inline GameDataParser/Program.cs:47]: Do we no longer need NpcParser?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: GameDataParser/Parsers/ItemParser.cs (hunk 1 above)
Comment (angelotadeucci) [reply GameDataParser/Parsers/ItemParser.cs:214]: I think the correct solution here is a regex expression. But I managed the desired results with:
`int parameter = int.Parse(rawParameter.FirstOrDefault(x => char.IsDigit(x)).ToString());`
Results:
https://gist.github.com/angelotadeucci/deb88b118a2e5b7503f39d02ef0855f1

Comment (Zintixx) [reply GameDataParser/Parsers/ItemParser.cs:214]: Egh...looks like the expiration date is in some of them.

**Review by AlanMorel — APPROVED** [review]

**Review by angelotadeucci — APPROVED** [review]

**Pull request merged**